### PR TITLE
Disable Jest addon for source files

### LINKED ISSUE #31: Jest addon should only work inside test/
See: https://github.com/reactjs/redux/blob/de94043033fdd13dc4fa3cede2943143c6d28942/.eslintrc.js#L4

### PR DESCRIPTION
**Pull request by acostalima**
Fixes #31

### CODE REVIEW

File: addons/jest.js [hunk 1]
```diff
@@ -10,4 +10,23 @@ module.exports = {
     'plugins': [
         'jest',
     ],
+    'overrides': [{
+        'env': {
+            'jest/globals': false,
+        },
+        'files': [
```
Comment (acostalima) [inline addons/jest.js:19]: Check if this key is needed

File: addons/jest.js [hunk 2]
```diff
@@ -10,4 +10,23 @@ module.exports = {
...
+            '**/*.js',
+        ],
+        'excludedFiles': [
+            '**/__tests__/**/*.js',
+            '**/?(.*)*@(spec|test).js',
+            'test/**/*',
```
Comment (acostalima) [inline addons/jest.js]: Include `test` and `tests`

File: addons/jest.js [hunk 3]
```diff
@@ -10,4 +10,23 @@ module.exports = {
...
+        ],
+        'rules': {
```
Comment (acostalima) [inline addons/jest.js]: Get the rules defined in `rules/jest.js` and apply a map operation to deactivate them all

File: test/fixtures/samples/jest/bad.file.name.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// `no-disabled-tests` - Disallow disabled tests
```
Comment (acostalima) [inline test/fixtures/samples/jest/bad.file.name.js]: Change file name to `non-test-file.js`

File: test/fixtures/samples/jest/__tests__/jest.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// `no-disabled-tests` - Disallow disabled tests
```
Comment (acostalima) [inline test/fixtures/samples/jest/__tests__/jest.js:1]: Add `test` and `tests` folder

File: test/.eslintrc.json [hunk 6]
```diff
@@ -1,8 +1,6 @@
 {
-    "root": true,
```
Comment (acostalima) [inline test/.eslintrc.json]: Revert this change

File: addons/jest.js (hunk 3 above)
Comment (satazor) [reply addons/jest.js]: https://github.com/jest-community/eslint-plugin-jest/blob/80c9b33e5e44529afb01d38c392b17bb1deee0ea/index.js#L65

File: .eslintrc.json [hunk 7]
```diff
@@ -2,7 +2,8 @@
     "root": true,
     "extends": [
         "./es5.js",
-        "./addons/node.js"
```
Comment (satazor) [inline .eslintrc.json]: Rever this change

**Review by satazor — CHANGES_REQUESTED** [review]

File: addons/jest.js (hunk 1 above)
Comment (acostalima) [reply addons/jest.js:19]: ESLint complains with an error saying `'files'` is mandatory

**Review by satazor — APPROVED** [review]

**Review by satazor — APPROVED** [review]
